hledger/test/balance/balance.test

```diff
@@ -172,27 +171,17 @@ $ hledger -f - balance -NE --declared --tree --depth 2 a
                    0  a
                    0    aa
                    0    ab
+                   0    ac
 
-# ** 11. In list mode we can see that non-leaf declared accounts are excluded.
-<
-account a
-account a:aa
-account a:ab
-account a:ac:aca
-account b
+# ** 11. In flat mode we can see that undeclared accounts are excluded.
```

Comment:
Text trivia:

I have been trying to move in the direction of "list" actually as the preferred term actually (for the -l short flag, eg. Possibly should go all the way and add --list long flag ?)

But your change matches the flag used in the command, so maybe it's better..

I feel maybe the old test description here was bad.. I think --declared includes declared accounts, rather than excluding undeclared ones. Perhaps we should drop that text.

Reply:
I'm ok with the behaviour change here personally. I'd put a ! (= "breaking change") after the first commit message prefix which helps me notice these at changelog/relnotes time. 

ref: is a new prefix meaning "refactoring" I guess. Ok, but perhaps it isn't the right one here since this is not strictly a refactoring. In such cases I often go with a generic dev:.

Reply:
I don't think it's new: I got it from the suggested prefixes in the developer documentation https://hledger.org/COMMITS.html. But you're correct that it's not just a refactoring, so I'll change to dev.

Reply:
I've modified the test descriptions and marked this as a breaking change.